File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.java
class Solution {
    public long appealSum(String s) {
        int last[] = new int[26];
        Arrays.fill(last, -1);
        long res = 0, n = s.length();
        for (int i = 0; i < s.length(); ++i) {
            res += (i - last[s.charAt(i) - 'a']) * (n - i);
            last[s.charAt(i) - 'a'] = i;
        }
        return res;
    }
}

File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.cpp
class Solution {
public:
   long long appealSum(string s) {
        vector<int> last(26, -1);
        long res = 0, n = s.size();
        for (int i = 0; i < n; ++i) {
            res += (i - last[s[i] - 'a']) * (n - i);
            last[s[i] - 'a'] = i;
        }
        return res;        
    }
};